kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,657 @@
+
+// Project includes
+#include "includes/define.h"
+#include "solving_strategies/strategies/residualbased_newton_raphson_strategy.h"
+#include "solving_strategies/convergencecriterias/convergence_criteria.h"
+#include "utilities/builtin_timer.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class ArcLengthStrategy
+ * @ingroup KratosCore
+ * @brief This is the base ArcLengthStrategy
+ * @details The theoretical details can be found in "Geometrical interpretation of the ARC-LENGTH method", M. Fafard and B. Massicotte,
+ * Computers and Structures Vol 46 pp 603-615 (1993). (Ramm arc-length method)
+ * @author Alejandro Cornejo and Ignasi de Pouplana
+ */
+template <class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class ArcLengthStrategy
+    : public ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+    public:
+    ///@name Type Definitions
+    ///@{
+
+    // Counted pointer of ClassName
+    KRATOS_CLASS_POINTER_DEFINITION(ArcLengthStrategy);
+
+    typedef ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+    typedef typename BaseType::TBuilderAndSolverType TBuilderAndSolverType;
+    typedef typename BaseType::TDataType TDataType;
+    typedef TSparseSpace SparseSpaceType;
+    typedef typename BaseType::TSchemeType TSchemeType;
+    typedef typename BaseType::TConvergenceCriteriaType TConvergenceCriteriaType;
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Constructor
+    ArcLengthStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        Parameters ThisParameters
+        ) : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme,
+                pNewConvergenceCriteria, pNewBuilderAndSolver, ThisParameters)
+        {
+            mDesiredIterations = ThisParameters["advanced_settings"]["desired_iterations"].GetInt();
+            mMaxRadiusFactor   = ThisParameters["advanced_settings"]["max_radius_factor"].GetDouble();
+            mMinRadiusFactor   = ThisParameters["advanced_settings"]["min_radius_factor"].GetDouble();
```

Comment:
BTW this should be implemented a bit differently, see here:
https://github.com/KratosMultiphysics/Kratos/blob/f180cdd09e6426b228df56f7c3424f0b1d1d9ee2/kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h#L137-L139